its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/CppTest.java

```diff
@@ -45,92 +45,82 @@
 class CppTest {
 
-  @TempDir
-  public Path basePath;
-
   @Test
   void testCppOnly() throws Exception {
-    String projectKey = "cpp";
-    String fileKey = "cpp:ConsoleApp/ConsoleApp.cpp";
-
-    ORCHESTRATOR.getServer().provisionProject(projectKey, "Cpp");
-    ORCHESTRATOR.getServer().associateProjectToQualityProfile(projectKey, "cpp", QualityProfiles.CPP_S106);
+    var context = AnalysisContext.forServer("CppSolution");
 
-    Path projectDir = TestUtils.projectDir(basePath, "CppSolution");
-    File wrapperOutDir = new File(projectDir.toFile(), "out");
+    ORCHESTRATOR.getServer().provisionProject(context.projectKey, context.projectKey);
+    ORCHESTRATOR.getServer().associateProjectToQualityProfile(context.projectKey, "cpp", QualityProfiles.CPP_S106);
 
-    String token = TestUtils.getNewToken(ORCHESTRATOR);
+    File wrapperOutDir = new File(context.projectDir.toFile(), "out");
 
-    TestUtils.newScannerBegin(ORCHESTRATOR, projectKey, projectDir, token)
+    var beginResult = context.begin
       .setProperty("sonar.cfamily.build-wrapper-output", wrapperOutDir.toString())
       .execute(ORCHESTRATOR);
-
-    File buildWrapperZip = new File(basePath.toString(), "build-wrapper-win-x86.zip");
-    File buildWrapperDir = basePath.toFile();
-    FileUtils.copyURLToFile(new URL(ORCHESTRATOR.getServer().getUrl() + "/static/cpp/build-wrapper-win-x86.zip"), buildWrapperZip);
-    ZipUtils.unzip(buildWrapperZip, buildWrapperDir);
+    assertThat(beginResult.isSuccess()).describedAs("C++ begin step failed with logs %n%s", beginResult.getLogs()). isTrue();
 
     String platformToolset = System.getProperty("msbuild.platformtoolset", "v140");
     String windowsSdk = System.getProperty("msbuild.windowssdk", "10.0.18362.0");
-
-    TestUtils.runMSBuildWithBuildWrapper(ORCHESTRATOR, projectDir, new File(buildWrapperDir, "build-wrapper-win-x86/build-wrapper-win-x86-64.exe"),
-      wrapperOutDir, "/t:Rebuild",
-      String.format("/p:WindowsTargetPlatformVersion=%s", windowsSdk),
-      String.format("/p:PlatformToolset=%s", platformToolset));
-
-    BuildResult result = TestUtils.executeEndStepAndDumpResults(ORCHESTRATOR, projectDir, projectKey, token);
-    assertThat(result.isSuccess()).isTrue();
-    assertThat(result.getLogs()).doesNotContain("Invalid character encountered in file");
-
-    List<Issue> issues = TestUtils.projectIssues(ORCHESTRATOR, projectKey);
-
-    List<String> keys = issues.stream().map(Issue::getRule).collect(Collectors.toList());
-    assertThat(keys).containsAll(List.of("cpp:S106"));
-
-    assertThat(TestUtils.getMeasureAsInteger(projectKey, "ncloc", ORCHESTRATOR)).isEqualTo(15);
-    assertThat(TestUtils.getMeasureAsInteger(fileKey, "ncloc", ORCHESTRATOR)).isEqualTo(8);
+    try (var buildWrapperDir = getBuildWrapperDir(context)) {
+      TestUtils.runMSBuildWithBuildWrapper(ORCHESTRATOR, context.projectDir, buildWrapperDir.path.resolve("build-wrapper-win-x86/build-wrapper-win-x86-64.exe").toFile(),
```

Comment:
Please remove this one from TestUtils and move it here, as a private method. It will never be reused anywhere else than in this CPP
```suggestion
      runMSBuildWithBuildWrapper(ORCHESTRATOR, context.projectDir, buildWrapperDir.path.resolve("build-wrapper-win-x86/build-wrapper-win-x86-64.exe").toFile(),
```

Reply:
This depends on GeneralCommand introduced in #2440